File: lib/nodejs/opentoken/src/tokentransformer/NoOperationTokenTransformer.ts
/**
 * Copyright (c) Truveta. All rights reserved.
 */

import { TokenTransformer } from './TokenTransformer';

/**
 * A **No Operation** token transformer. No transformation is applied whatsoever.
 */
export class NoOperationTokenTransformer implements TokenTransformer {
  /**
   * No operation token transformer.
   *
   * Does not transform the token in any ways.
   *
   * @param token - The token to transform.
   * @returns the same token, unchanged.
   */
  transform(token: string): string {
    return token;
  }
}

File: lib/python/opentoken/src/main/opentoken/tokentransformer/no_operation_token_transformer.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from opentoken.tokens.token import Token
from opentoken.tokentransformer.token_transformer import TokenTransformer


class NoOperationTokenTransformer(TokenTransformer):
    """
    A No Operation token transformer. No transformation is
    applied whatsoever.
    """

    def transform(self, token: str) -> str:
        """
        No operation token transformer.

        Does not transform the token in any ways.

        Args:
            token: The token to be transformed.

        Returns:
            The unchanged token.
        """
        if token is None or (isinstance(token, str) and token.strip() == ""):
            token = Token.BLANK  # Return blank token for null or blank input
        return token
